File: simple/src/main/java/com/example/simple/MainActivity.java
package com.example.simple;

import android.Manifest;
import android.os.Build;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import android.view.View;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

import cn.yzl.imgpicker.ImgPicker;
import cn.yzl.imgpicker.PickerResult;

public class MainActivity extends AppCompatActivity {

    ImgPicker rxImagePicker;
    ImageView iv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        iv = findViewById(R.id.iv);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            requestPermissions(new String[]{Manifest.permission.CAMERA,
                    Manifest.permission.READ_EXTERNAL_STORAGE}, 100);
        }
        rxImagePicker = new ImgPicker(this, ImgPicker.DEFAULT_OPTION.clone().setCrop(true)
                .setRootDir(this.getCacheDir().getAbsolutePath()))
                .success(new ImgPicker.SucessCallBack() {

                    @Override
                    public void sucess(PickerResult ret) {
                        Glide.with(MainActivity.this)
                                .load(ret.uri)
                                .into(iv);
                    }
                });

        findViewById(R.id.bt_camera)
                .setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        rxImagePicker.openCamera();
                    }
                });

        findViewById(R.id.bt_album)
                .setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        rxImagePicker.openAlbum();
                    }
                });
    }

}

File: simple/src/main/java/com/example/simple/MainActivity.kt
package com.example.simple

import android.Manifest
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import cn.yzl.imgpicker.ImgPicker
import cn.yzl.imgpicker.PickerResult
import com.bumptech.glide.Glide

class MainActivity : AppCompatActivity() {

    lateinit var rxImagePicker: ImgPicker
    var iv: ImageView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        iv = findViewById(R.id.iv)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            requestPermissions(
                    arrayOf(
                            Manifest.permission.CAMERA,
                            Manifest.permission.READ_EXTERNAL_STORAGE
                    ), 100
            )
        }
        rxImagePicker = ImgPicker(
                this,
                crop = true,
                rootDir = this.cacheDir.absolutePath)
                .success {
                    Glide.with(this@MainActivity)
                            .load(it!!.uri)
                            .into(iv)
                }
        findViewById<View>(R.id.bt_camera)
                .setOnClickListener { rxImagePicker.openCamera() }
        findViewById<View>(R.id.bt_album)
                .setOnClickListener { rxImagePicker.openAlbum() }
    }
}